src/idl/include/idl/tree.h

```diff
@@ -341,6 +346,14 @@ struct idl_struct {
   idl_extensibility_t extensibility;
 };
 
+typedef struct idl_forward_decl idl_forward_decl_t;
+struct idl_forward_decl {
```

Comment:
I think I'd liked it better if the forward contained a `type_spec` member that's populated when the actual type is parsed.

Reply:
In 4824b66 I've added a field `definition` in `idl_forward_t` that gets the `type_spec`